File: datastructure using cpp/binarySearch/binarySearch.py
def binarySearch(arr, key):
    l = 0
    h = len(arr)-1
    while(l<=h):
        mid = (l+h)//2
        if arr[mid]==key:
            return mid + 1
        elif arr[mid]>key:
            h = mid - 1
        else:
            l = mid+1
    return 0

arr = [3,6,8,12,14,17,25,29,31,36,42,47,53,55,6]
key = 3

print(binarySearch(arr,key))

File: graph/BinarySearch.java
public class BinarySearch {
 
    public static void main(String[] args) {
 
        int[] a = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11};
        System.out.println(binarySearchIterative(a, 8));
        System.out.println(binarySearchRecursive(a, 0, a.length - 1, 8));
 
    }
 
    public static int binarySearchIterative(int[] a, int key) {
 
        int lo = 0;
        int hi = a.length - 1;
 
        while (lo <= hi) {
            int mid = lo + (hi - lo) / 2;
            if (key < a[mid]) {
                hi = mid - 1;
            } else if (key > a[mid]) {
                lo = mid + 1;
            } else {
                return mid;
            }
        }
 
        return -1;
    }
 
    private static int binarySearchRecursive(int[] a, int lo, int hi, int key) {
 
        if (hi < lo) {
            return -1;
        }
 
        int mid = (lo + hi) / 2;
 
        if (key == a[mid]) {
            return mid;
        } else if (key < a[mid]) {
            return binarySearchRecursive(a, lo, mid - 1, key);
        } else {
            return binarySearchRecursive(a, mid + 1, hi, key);
        }
 
    }
 
}